cedar-lean/Cedar/Thm/Data/Map.lean

```diff
@@ -85,59 +224,128 @@ theorem make_mem_list_mem [LT α] [StrictLT α] [DecidableLT α] {xs : List (α
   simp only [kvs, make]
   intro h₁
   have h₂ := List.canonicalize_subseteq Prod.fst xs
-  simp [List.subset_def] at h₂
+  simp only [List.subset_def] at h₂
   exact h₂ h₁
 
+/--
+  Very similar to `make_mem_list_mem` above
+-/
+theorem mem_values_make [LT α] [StrictLT α] [DecidableLT α] {xs : List (α × β)} :
+  v ∈ (Map.make xs).values → v ∈ xs.map Prod.snd
+:= by
+  -- despite the similarity to `make_mem_list_mem`, the proof does not currently
+  -- use `make_mem_list_mem`
+  simp only [values, make]
+  simp only [List.mem_map, forall_exists_index, and_imp]
+  intro (k, v) h₁ h₂
+  exists (k, v)
+  subst h₂
+  simp only [and_true]
+  have h₂ := List.canonicalize_subseteq Prod.fst xs
+  simp only [List.subset_def] at h₂
+  exact @h₂ (k, v) h₁
+
 theorem make_nil_is_empty {α β} [LT α] [DecidableLT α] :
   (Map.make [] : Map α β) = Map.empty
 := by simp [make, empty, List.canonicalize_nil]
 
+/--
+  Note that the converse of this is not true:
+  counterexample `xs = [(1, false)]`, `ys = []`, `ab = (1, false)`.
+-/
+theorem make_cons [LT α] [DecidableLT α] {xs ys : List (α × β)} {ab : α × β} :
```

Comment:
Maybe change to `make_cons_pair` to distinguish from `make_cons` in Set.lean?

Reply:
This one is `Map.make_cons` and that one is `Set.make_cons`, is that enough to distinguish?  They're basically the same theorem but for `Map.make` and `Set.make` respectively